src/snapred/backend/recipe/algorithm/AlignAndFocusReductionAlgorithm.py

```diff
@@ -47,9 +50,16 @@ def PyExec(self):
         )
 
         self.mantidSnapper.executeQueue()
-        DMin = reductionIngredients.reductionState.stateConfig.focusGroups[0].dMin
-        DMax = reductionIngredients.reductionState.stateConfig.focusGroups[0].dMax
-        DeltaRagged = reductionIngredients.reductionState.stateConfig.focusGroups[0].dBin
+        dMinList = []
+        dMaxList = []
+        dBinList = []
+        for i in pixelGroupingParameters:
+            dMinList.append(i.dResolution.minimum)
+            dMaxList.append(i.dResolution.maximum)
+            dBinList.append(i.dRelativeResolution)
```

Comment:
I strongly recommend making these dictionaries keyed to the group ID:
``` python
dMin = {pgp.groupID:grp.dResolution.minimum for pgp in pixelGroupingParameters}
dMax = {pgp.groupID:grp.dResolution.maximum for pgp in pixelGroupingParameters}
dBin = {pgp.groupID:grp.dRelativeResolution / numberOfBins for pgp in pixelGroupingParameters}
```
where `numberOfBins` is the number of bins per peak we were wanting to use